go/vt/vtgate/semantics/scoper.go

```diff
@@ -184,6 +188,9 @@ func (s *scoper) enterJoinScope(cursor *sqlparser.Cursor) {
 
 func (s *scoper) pushSelectScope(node *sqlparser.Select) {
 	currScope := newScope(s.currentScope())
+	if len(s.scopes) > 0 && s.scopes[len(s.scopes)-1] != s.currentScope() {
+		fmt.Println("BUG: scope counts did not match")
+	}
```

Comment:
looks like some testing print, if this check is needed, then it should return an error